src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Services/DatalockValidationService.cs

```diff
@@ -0,0 +1,295 @@
+                            x.AimSeqNumber == earning.AimSeqNumber &&
+                            x.PriceEpisodeIdentifier == earning.PriceEpisodeIdentifier &&
+                            x.RuleId == error &&
+                            x.Period == earning.Period &&
+                            x.Ukprn == earning.Ukprn) == null)
+                    {
+                        ValidationErrorsByPeriod.Add(new DatalockValidationErrorByPeriod()
+                        {
+                            LearnRefNumber = earning.LearnRefNumber,
+                            AimSeqNumber = earning.AimSeqNumber,
+                            PriceEpisodeIdentifier = earning.PriceEpisodeIdentifier,
+                            RuleId = error,
+                            Ukprn = earning.Ukprn,
+                            Period = earning.Period,
+                        });
+                    }
+                }
+            }
+            else
+            {
+                payable = true;
+            }
+
+            PriceEpisodePeriodMatches.Add(new PriceEpisodePeriodMatchEntity
+            {
+                AimSeqNumber = earning.AimSeqNumber,
+                CommitmentId = commitment.CommitmentId,
+                LearnRefNumber = earning.LearnRefNumber,
+                PriceEpisodeIdentifier = earning.PriceEpisodeIdentifier,
+                Period = earning.Period,
+                TransactionTypesFlag = paymentType,
+                Payable = payable,
+                Ukprn = earning.Ukprn,
+                VersionId = commitment.VersionId,
+            });
+
+            if (payable)
+            {
+                var validationErrorsToRemove = ValidationErrors.Where(x =>
+                        x.Ukprn == earning.Ukprn &&
+                        x.LearnRefNumber == earning.LearnRefNumber &&
+                        x.PriceEpisodeIdentifier == earning.PriceEpisodeIdentifier)
+                    .ToList();
+
+                foreach (var validationError in validationErrorsToRemove)
+                {
+                    ValidationErrors.Remove(validationError);
+                }
+            }
+
+            if (PriceEpisodeMatches.FirstOrDefault(x =>
+                    x.IsSuccess == payable &&
+                    x.CommitmentId == commitment.CommitmentId &&
+                    x.LearnRefNumber == earning.LearnRefNumber &&
+                    x.PriceEpisodeIdentifier == earning.PriceEpisodeIdentifier &&
+                    x.Ukprn == earning.Ukprn) == null)
+            {
+                PriceEpisodeMatches.Add(new PriceEpisodeMatchEntity
+                {
+                    CommitmentId = commitment.CommitmentId,
+                    IsSuccess = payable,
+                    LearnRefNumber = earning.LearnRefNumber,
+                    PriceEpisodeIdentifier = earning.PriceEpisodeIdentifier,
+                    Ukprn = earning.Ukprn,
+                });
+            }
+        }
+    }
+}
```

Comment:
It's a very big method and it could be simplified by splitting out some of the code.